File: python_programming_basics/nested_loops/e_04_train_the_trainers.py
jury_count = int(input())
presentation = input()
total_average_grade = 0
presentation_count = 0

while presentation != "Finish":
    presentation_average = 0

    for _ in range(jury_count):
        current_grade = float(input())
        presentation_average += current_grade

    presentation_average /= jury_count
    total_average_grade += presentation_average
    presentation_count += 1

    print(f"{presentation} - {presentation_average:.2f}.")

    presentation = input()

final_average_grade = total_average_grade / presentation_count
print(f"Student's final assessment is {final_average_grade:.2f}.")

File: JavaScript Basics/06. Nested Loops/e-04-Train-The-Trainers.js
function trainTheTrainers(input) {
    let index = 0;
    const juryCount = Number(input[index]);
    index++;

    let totalSum = 0;
    let totalGrades = 0;

    while (input[index] !== "Finish") {
        let presentationName = input[index];
        index++;

        let sumGrades = 0;

        for (let i = 0; i < juryCount; i++) {
            let grade = Number(input[index]);
            index++;
            sumGrades += grade;
        }

        let averageGrade = sumGrades / juryCount;
        totalSum += sumGrades;
        totalGrades += juryCount;

        console.log(`${presentationName} - ${averageGrade.toFixed(2)}.`);
    }

    let finalAssessment = totalSum / totalGrades;
    console.log(`Student's final assessment is ${finalAssessment.toFixed(2)}.`);
}


trainTheTrainers([
    "2",
    "While-Loop",
    "6.00",
    "5.50",
    "For-Loop",
    "5.84",
    "5.66",
    "Finish"
]);

trainTheTrainers([
    "3",
    "Arrays",
    "4.53",
    "5.23",
    "5.00",
    "Lists",
    "5.83",
    "6.00",
    "5.42",
    "Finish"
]);

trainTheTrainers([
    "2",
    "Objects and Classes",
    "5.77",
    "4.23",
    "Dictionaries",
    "4.62",
    "5.02",
    "RegEx",
    "2.88",
    "3.42",
    "Finish"
]);
